robot_name/robot_name.rb

```diff
@@ -1,26 +1,35 @@
 class NameCollisionError < RuntimeError; end
 
 class Robot
+
+
   attr_accessor :name
 
-  @@registry
 
-  def initialize(args = {})
+  def initialize
     @@registry ||= []
-    @name_generator = args[:name_generator]
+    name_generator
+    check_registry
+  end
 
-    if @name_generator
-      @name = @name_generator.call
-    else
-      generate_char = -> { ('A'..'Z').to_a.sample }
-      generate_num = -> { rand(10) }
+  def name_generator
+    char = -> { ('A'..'Z').to_a.sample }
+    num = -> { rand(10) }
+    self.name = "#{char.call}#{char.call}#{num.call}#{num.call}#{num.call}"
+  end
 
-      @name = "#{generate_char.call}#{generate_char.call}#{generate_num.call}#{generate_num.call}#{generate_num.call}"
+  def check_registry
+    if !(name =~ /[[:alpha:]]{2}[[:digit:]]{3}/) || @@registry.include?(name)
+     raise NameCollisionError, 'There was a problem generating the robot name!'
```

Comment:
`NameCollisionError` is used for both name collision and invalid name (which isn't a collision). Should probably be 2 different error classes.